### DISCUSSION TITLE
vendor folder

### DISCUSSION THREAD

**Discussion by Fredrick-KE** [PHPMailer/PHPMailer#2283] [General]
I haven't seen the vendor folder when trying to clone the project!

**Comment by Synchro** [comment]
That's because there is no vendor folder when you clone the project, and nor should there be one.

The vendor folder is created by composer when you add PHPMailer to your own project (if you're **using** PHPMailer) or if you clone it and then run `composer install` (if you're **working on** PHPMailer itself).

Please refer to the installation instructions in the readme, and composer's documentation.

**Comment by Fredrick-KE** [comment]
How do I go about that? Am a junior developer !

**Reply by Synchro** [reply]
First rule of being a junior developer: read the docs. The installation instructions for PHPMailer are [here](https://github.com/PHPMailer/PHPMailer#installation--loading), and the composer documentation is [here](https://getcomposer.org/doc/00-intro.md).

**Reply by Fredrick-KE** [reply]
Thank you

**Comment by Fredrick-KE** [comment]
I have installed composer in my pc, what else should I do? its not loading
the vendor folder in the project !

**Reply by Synchro** [reply]
What happened when you ran `composer require phpmailer/phpmailer` in your project folder, as the readme tells you to?

**Comment by Fredrick-KE** [comment]
This is my composer.json code where should I place the *composer require
Phpmailer/Phpmailer*



{
    "name": "phpmailer/phpmailer",
    "type": "library",
    "description": "PHPMailer is a full-featured email creation and
transfer class for PHP",
    "authors": [
        {
            "name": "Fredrick",
            "email": ***@***.***"
        },
        {
            "name": "Fred",
            "email": ***@***.***"
        },
        {
            "name": "Andy Prevost",
            "email": ***@***.***"
        },
        {
            "name": "Code_with_fredrick"
        }
    ],
    "funding": [
        {
            "url": "https://github.com/fredy-ricky",
            "type": "github"
        }
    ],
    "require": {
        "php": ">=5.5.0",
        "ext-ctype": "*",
        "ext-filter": "*",
        "ext-hash": "*"
    },
    "require-dev": {
        "dealerdirect/phpcodesniffer-composer-installer": "^0.7.0",
        "doctrine/annotations": "^1.2",
        "phpcompatibility/php-compatibility": "^9.3.5",
        "roave/security-advisories": "dev-latest",
        "squizlabs/php_codesniffer": "^3.5.6",
        "yoast/phpunit-polyfills": "^0.2.0"
    },
    "suggest": {
        "ext-mbstring": "Needed to send email in multibyte encoding
charset",
        "hayageek/oauth2-yahoo": "Needed for Yahoo XOAUTH2 authentication",
        "league/oauth2-google": "Needed for Google XOAUTH2 authentication",
        "psr/log": "For optional PSR-3 debug logging",
        "stevenmaguire/oauth2-microsoft": "Needed for Microsoft XOAUTH2
authentication",
        "symfony/polyfill-mbstring": "To support UTF-8 if the Mbstring PHP
extension is not enabled (^1.2)"
    },
    "autoload": {
        "psr-4": {
            "PHPMailer\\PHPMailer\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "PHPMailer\\Test\\": "test/"
        }
    },
    "license": "LGPL-2.1-only"
}

**Reply by Synchro** [reply]
Nope, you're doing it wrong. This is not the composer.json file for your own project, it's PHPMailer's own, which you do not need to touch. The `composer install...` line is a command you run in a terminal, you don't put it in a file. If you were starting a new project, you'd do this:

    mkdir my_ project
    cd my_project
    composer require phpmailer/phpmailer

That would create a `composer.json` file, a `composer.lock` file, a `vendor` folder, install PHPMailer in it, and create an `autoload.php` file to load it with. The whole point of composer is that you do not have to touch your libraries, and you should not anyway!

**Comment by Fredrick-KE** [comment]
Thanks for the tip!
